tests/test_dispersion.py

```diff
@@ -0,0 +1,125 @@
+import io
+import sys
+
+import generate_tempfiles
+import numpy as np
+from calcos import dispersion
+
+
+def test_init():
+    # setup
+    # create a temp disptab
+    # this function is only used by the functions in this script.
+    tmp_file = "test_disptab_init.fits"
+    generate_tempfiles.create_disptab_file(tmp_file)
+    t = "49g17153l_disp.fits"
+    filters = {'FPOFFSET': 0,
+               'NELEM': 2
+               }
+    disp_obj = dispersion.Dispersion(t, filters, True)
+
+    return disp_obj
+
+
+def test_info():
+    # Setup
+    disp_obj = test_init()
+    captured_msg = io.StringIO()
+    sys.stdout = captured_msg
+    expected_str = "filter = {'nelem': 2}\n" \
+                   "use_fpoffset = True\n" \
+                   "fpoffset = 0\n" \
+                   "number of coefficients = 2\n" \
+                   "coeff = [1.04502424e+03 9.95559148e-03]\n" \
+                   "delta = 0\n" \
+                   "number of matching rows = 78\n" \
+                   "valid = True\n"
+    # Test
+    disp_obj.info()
+    sys.stdout = sys.__stdout__
+    captured_msg = captured_msg.getvalue()
+    # Verify
+    assert expected_str == captured_msg
+
+
+def test_isValid():
+    # Setup
+    disp_obj = test_init()
+    expected = True
+    # Test
+    test = disp_obj.isValid()
+    # Verify
+    assert expected == test
+
+
+def test_getNRows():
+    # Setup
+    disp_obj = test_init()
+    expected_val = 78
+    # Test
+    num_of_rows = disp_obj.getNRows()
+    # Verify
+    assert expected_val == num_of_rows
+
+
+def test_getFilter():
+    # Setup
+    disp_obj = test_init()
+    expected_filter = {'nelem': 2}
+    # Test
+    actual_filter = disp_obj.getFilter()
+    # Verify
+    assert expected_filter == actual_filter
+
+
+def test_close():
+    # Setup
+    disp_obj = test_init()
+    expected_filter = {}
+    exp_ncoeff = 0
+    exp_delta = 0.
+    exp_fpoffset = 0
+    exp_nrows = 0
+    exp_valid = False
```

Comment:
It might be good to test non-zero values for these as well if that is an easy change.